### PR TITLE
Improve charts accessibility

### PR DESCRIPTION
**Pull request by Aljullu**
This PR implements most (if not all) changes suggested in #208.

* #339 made it possible to focus columns (dates) in charts, this PR makes it possible to focus every single point (in line charts) and every single bar (in bar charts).

* Each point/bar has an `aria-label` property used by screen readers with the date, the label and the value.

* The Y and X axis are hidden. Given that the `aria-label` of each point already includes that information, we don't need them to be visible to screen readers.

* Functions `handleMouseOverBarChart`, `handleMouseOutLineChart`, etc. have been modified a bit, so we pass them fewer parameters. And I also removed some `mouseover`, `mouseout` events that were never triggered.

**Screenshot**
image

**Steps to test**
* Activate the Screen Reader in your system. That depends on the operating system you're using, in Gnome Linux distros it is under _Settings_ > _Universal access_.
* Go to any page with a chart.
* Navigate the page with the `Tab` key.
* Verify that you can focus chart elements and, when focused, the tooltip is displayed.
* Verify that the screen reader reads out the chart information.
* Check both, the line chart and the bar chart.

**Future work**
* The buttons to change between line chart and bar chart are not accessible (#422).
image
* For now I'm using the tooltip date for the screen reader, but with #373 I understand it will disappear. So a new param will need to be created (or renaming the `tooltipFormat` one).

Disclaimer: I'm not very experienced in accessibility, so feel free to do a harsh review and point me out any issue you encounter! :slightly_smiling_face:

### CODE REVIEW

**Review by timmyc — COMMENTED** [review]
Nice work here! I can navigate the chart tabs and voiceover reads out the active data point for me. The only odd thing I noticed is when on a point in the line chart, it would say "You are on an image" - wonder if we should have it say something else, like you are on a point in the chart perhaps?

I'll let @ryelle give this some 👀 too

File: client/components/chart/utils.js
Comment (ryelle) [inline client/components/chart/utils.js]: Having `key` in this label makes the label too long & a bit confusing on reports, for example I hear: `August 06 2018 Last Month (Aug 1 - 31, 2018) $8,080.81` – it's unclear what that dollar value represents, the entire month or just that day? Since each line is in a region with the key label, I think we can omit it here.

File: client/components/chart/utils.js
Comment (ryelle) [inline client/components/chart/utils.js]: I have the same problem as above, the key is extra info & can be omitted

Comment (ryelle) [reply client/components/chart/utils.js]: I see/hear that this is a problem with the previous period, since `getTooltipDate( params, d.date )` will always get the "current" date, not the date of the previous period's value. Is there a way to change that? So that if I'm comparing August 1-31 vs July 1-31, it can tell me `August 6th, 2018 $500` and then `July 6th, 2018 $450`?

Currently it would say `August 6 2018 Last Month (Aug 1 - 31, 2018) $500`, then `August 6 2018 Previous Period (Jul 1 - 31, 2018) $450`.

**Review by ryelle — COMMENTED** [review]
This is a good start on screen reader compatibility 👍  I have a few comments on what the labels themselves should be, but moving through with VoiceOver worked fairly well.

I did notice that the x-axis is not hidden, so I still need to navigate through that before getting into the chart content.

**Comment by timmyc** [comment]
@Aljullu do you need another review pass on this PR?

**Comment by Aljullu** [comment]
> @Aljullu do you need another review pass on this PR?

@timmyc not yet! I'm still working on this comment from @ryelle (https://github.com/woocommerce/wc-admin/pull/421#discussion_r218251012).

The code needed to do that is very related to issue #373, so I'm working on both things together. I will ping you again when it's ready. :slightly_smiling_face:

**Comment by Aljullu** [comment]
@timmyc @ryelle I rebased the branch with the last changes in `master` and implemented the changes discussed in this PR.

Can you take a look when you have a moment?

**Comment by Aljullu** [comment]
Given that #467 adds some changes that will conflict with this branch, I will leave this one on hold until I merge that one and will ask your review again later.

**Comment by Aljullu** [comment]
It's ready to be reviewed again @timmyc @ryelle. :slightly_smiling_face:

**Review by ryelle — APPROVED** [review]
Overall this is a great improvement. I still think the exact labels need updating to be more clear, but that should be done in a follow-up PR after some discussion about what's useful (maybe we can set aside some time to talk about analytics accessibility at the GM team day @timmyc?)

**Comment by timmyc** [comment]
> maybe we can set aside some time to talk about analytics accessibility at the GM team day?

💯

**Pull request merged**
